.github/CONTRIBUTING.md

```diff
@@ -44,6 +78,49 @@ Once you're confident that your upcoming changes will be accepted:
 * If possible, submit tests along with your topic branch. It will help a lot to get your your patch / new feature accepted, and should prevent unwanted breaking changes to silently happen in future developments.
```

Comment:
We should update the wording here as tests being _required_ unless they are not applicable for some reason (like docs changes, dependency constraint updates, etc).